copycat/runtime/src/main/java/org/apache/kafka/copycat/runtime/rest/resources/ConnectorsResource.java

```diff
@@ -0,0 +1,163 @@
+    private static final long REQUEST_TIMEOUT_MS = 90 * 1000;
+
+    private final Herder herder;
+    @javax.ws.rs.core.Context
+    private ServletContext context;
+
+    public ConnectorsResource(Herder herder) {
+        this.herder = herder;
+    }
+
+    @GET
+    @Path("/")
+    public Collection<String> listConnectors() throws Throwable {
+        FutureCallback<Collection<String>> cb = new FutureCallback<>();
+        herder.connectors(cb);
+        return completeOrForwardRequest(cb, "/connectors", "GET", null, new TypeReference<Collection<String>>() { });
+    }
+
+    @POST
+    @Path("/")
+    public void createConnector(final CreateConnectorRequest createRequest) throws Throwable {
+        String name = createRequest.name();
+        Map<String, String> configs = createRequest.config();
+        if (!configs.containsKey(ConnectorConfig.NAME_CONFIG))
+            configs.put(ConnectorConfig.NAME_CONFIG, name);
+
+        FutureCallback<String> cb = new FutureCallback<>();
+        herder.addConnector(configs, cb);
+        completeOrForwardRequest(cb, "/connectors", "POST", createRequest);
+    }
+
+    @GET
+    @Path("/{connector}")
+    public ConnectorInfo getConnector(final @PathParam("connector") String connector) throws Throwable {
+        FutureCallback<ConnectorInfo> cb = new FutureCallback<>();
+        herder.connectorInfo(connector, cb);
+        return completeOrForwardRequest(cb, "/connectors/" + connector, "GET", null, new TypeReference<ConnectorInfo>() { });
+    }
+
+    @GET
+    @Path("/{connector}/config")
+    public Map<String, String> getConnectorConfig(final @PathParam("connector") String connector) throws Throwable {
+        FutureCallback<Map<String, String>> cb = new FutureCallback<>();
+        herder.connectorConfig(connector, cb);
+        return completeOrForwardRequest(cb, "/connectors/" + connector + "/config", "GET", null, new TypeReference<Map<String, String>>() { });
+    }
+
+    @PUT
+    @Path("/{connector}/config")
+    public void putConnectorConfig(final @PathParam("connector") String connector,
+                                   final Map<String, String> connectorConfig) throws Throwable {
+        FutureCallback<Void> cb = new FutureCallback<>();
+        herder.putConnectorConfig(connector, connectorConfig, cb);
+        completeOrForwardRequest(cb, "/connectors/" + connector + "/config", "PUT", connectorConfig, null);
+    }
+
+    @GET
+    @Path("/{connector}/tasks")
+    public List<TaskInfo> getTaskConfigs(final @PathParam("connector") String connector) throws Throwable {
+        FutureCallback<List<TaskInfo>> cb = new FutureCallback<>();
+        herder.taskConfigs(connector, cb);
+        return completeOrForwardRequest(cb, "/connectors/" + connector + "/tasks", "GET", null, new TypeReference<List<TaskInfo>>() { });
+    }
+
+    @POST
+    @Path("/{connector}/tasks")
+    public void putTaskConfigs(final @PathParam("connector") String connector,
```

Comment:
Great question. 

1. Right now, the REST API is the only mechanism for doing this (there was a FIXME that got resolved because this became available). We could potentially separate the two (e.g. on different ports), but the REST API is the only means of inter-worker communication beyond the assignment protocol.
2. I tend to use POST on the "collection" resources, where they may generate subresources as a result. I think this is just a matter of how you think of different resources. I'm open to using either one.